static/app/components/codeSnippet.tsx

```diff
@@ -48,71 +176,86 @@ export function CodeSnippet({
       : t('Unable to copy');
 
   return (
-    <Fragment>
-      {!hideActionBar && (
-        <CodeContainerActionBar>
-          {filename && <span>{filename}</span>}
-          <Tooltip delay={0} isHoverable={false} title={tooltipTitle} position="bottom">
-            <UnstyledButton
+    <Wrapper>
+      {filename && (
+        <Header>
+          {filename}
+          {!hideCopyButton && (
+            <CopyButton
               type="button"
+              size="xs"
+              borderless
+              isInHeader
               onClick={handleCopy}
+              title={tooltipTitle}
+              tooltipProps={{delay: 0, isHoverable: false, position: 'left'}}
               onMouseLeave={() => setTooltipState('copy')}
             >
-              <IconCopy />
-            </UnstyledButton>
-          </Tooltip>
-        </CodeContainerActionBar>
+              <IconCopy size="xs" />
+            </CopyButton>
+          )}
+        </Header>
       )}
 
-      <PreContainer unsetBorderRadiusTop={!hideActionBar}>
-        <code ref={ref} className={`language-${language}`}>
+      {!hideCopyButton && !filename && (
+        <CopyButton
+          type="button"
+          size="xs"
+          onClick={handleCopy}
+          title={tooltipTitle}
+          tooltipProps={{delay: 0, isHoverable: false, position: 'left'}}
+          onMouseLeave={() => setTooltipState('copy')}
+        >
+          <IconCopy size="xs" />
+        </CopyButton>
+      )}
```

Comment:
When the header is not rendered (there is no `filename`), the copy button will only appear on hover:
<img width="630" alt="Screen Shot 2022-09-30 at 12 12 33 PM" src="https://user-images.githubusercontent.com/44172267/193340627-4495f32e-ded1-4c98-988f-5fac12e44bea.png">
<img width="630" alt="Screen Shot 2022-09-30 at 12 12 40 PM" src="https://user-images.githubusercontent.com/44172267/193340639-c3a49ab7-a105-4b54-9311-de80cf62d358.png">

Reply:
This is a nice touch!